### PR TITLE
[ModelTiming] Corrected timing method

### PR DESCRIPTION
**Pull request by Efrainq07**
Corrected timing method in order to account for the fact that GPUs execute processes asynchronously. This is done as mentioned in this [article](https://deci.ai/the-correct-way-to-measure-inference-time-of-deep-neural-networks/).

Also added new flag --batchsize for specifying the batchsize for the training run.

### CODE REVIEW

**Comment by Efrainq07** [comment]
@blefaudeux @min-xu-ai

File: examples/python/pytorch/resnet152_DDP_training.py [hunk 1]
```diff
@@ -8,49 +8,52 @@
...
+    print("Running init_process_group...")
     dist.init_process_group("nccl", rank=rank, world_size=world_size)
+    print("Finished init_process_group...")
 
 def cleanup():
     dist.destroy_process_group()
 
-def train(local_rank, args):
-    rank = args.nr * args.gpus + local_rank	
+def train(gpu, args):
```
Comment (min-xu-ai) [inline examples/python/pytorch/resnet152_DDP_training.py:24]: the variable name "gpu" is vague. Perhaps rename to gpu_id?

File: examples/python/pytorch/resnet152_DDP_training.py [hunk 2]
```diff
@@ -8,49 +8,52 @@
...
+    for _ in range(10):
+        _ = model(dummy_input)
+    model = nn.parallel.DistributedDataParallel(model,device_ids=[gpu])
 
     criterion = nn.CrossEntropyLoss().cuda()
     optimizer = optim.SGD(model.parameters(), lr=0.001, momentum=0.9)
-
-    print("Starting training")
     training_run_data = pd.DataFrame(columns=['epoch','batch','batch_size','gpu_number','time'])
+    starter, ender = torch.cuda.Event(enable_timing=True),torch.cuda.Event(enable_timing=True)
```
Comment (min-xu-ai) [inline examples/python/pytorch/resnet152_DDP_training.py:50]: can you add a bit comment here to say what's the benefit of torch.cuda.Event over time.time()?

**Review by min-xu-ai — COMMENTED** [review]
looks good. some small comments.

File: examples/python/pytorch/resnet152_DDP_training.py [hunk 3]
```diff
@@ -8,49 +8,52 @@
...
-    torch.cuda.set_device(local_rank)
-    model.cuda()
 
-    model = nn.parallel.DistributedDataParallel(model,device_ids=[local_rank])
-    print("Setting optimizer")
+    model = models.resnet152()
+    torch.cuda.set_device(gpu)
+    model.cuda()
+    print("GPU initialization")
+    dummy_input = torch.randn(1, 3,224,224, dtype=torch.float).to(gpu)
```
Comment (msbaines) [inline examples/python/pytorch/resnet152_DDP_training.py:42]: Why is this needed?

File: examples/python/pytorch/resnet152_DDP_training.py (hunk 2 above)
Comment (msbaines) [inline examples/python/pytorch/resnet152_DDP_training.py:50]: I think you need to create a new event every time.

File: examples/python/pytorch/resnet152_DDP_training.py (hunk 1 above)
Comment (msbaines) [reply examples/python/pytorch/resnet152_DDP_training.py:24]: > the variable name "gpu" is vague. Perhaps rename to gpu_id?

https://medium.com/@auro_227/timing-your-pytorch-code-fragments-e1a556e81f2

File: examples/python/pytorch/resnet152_DDP_training.py (hunk 2 above)
Comment (msbaines) [reply examples/python/pytorch/resnet152_DDP_training.py:50]: > can you add a bit comment here to say what's the benefit of torch.cuda.Event over time.time()?

https://medium.com/@auro_227/timing-your-pytorch-code-fragments-e1a556e81f2

File: examples/python/pytorch/resnet152_DDP_training.py (hunk 3 above)
Comment (Efrainq07) [reply examples/python/pytorch/resnet152_DDP_training.py:42]: It is mentioned here that the performance of GPUs is different when they just started from when are already running. The dummy input there is what is done on this article:
https://deci.ai/the-correct-way-to-measure-inference-time-of-deep-neural-networks/
